pkg/xds/secrets/secrets_test.go

```diff
@@ -190,13 +202,34 @@ var _ = Describe("Secrets", Ordered, func() {
 
 		Context("should regenerate certificate", func() {
 			BeforeEach(func() {
-				_, _, err := secrets.GetForDataPlane(context.Background(), newDataplane(), newMesh(), nil)
+				_, _, err := secrets.GetForDataPlane(context.Background(), newDataplane(), newMesh("default"), nil)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It("when cross-mesh was added and previous removed", func() {
+				// given
+				defaultMesh := newMesh("default")
+
+				// when
+				_, _, err := secrets.GetForDataPlane(context.Background(), newDataplane(), defaultMesh, []*core_mesh.MeshResource{newMesh("mesh-1")})
+
+				// then
+				Expect(err).ToNot(HaveOccurred())
+				Expect(test_metrics.FindMetric(metrics, "cert_generation").GetCounter().GetValue()).To(Equal(1.0))
+				Expect(test_metrics.FindMetric(metrics, "cert_generation_other_meshes", "other_mesh", "mesh-1").GetCounter().GetValue()).To(Equal(1.0))
```

Comment:
as mentioned above - instead of asserting on the metric is it possible to assert on something else? or maybe we should move `shouldGenerateCerts` to a separate module, make it public and test it there? 🤔